x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/dataframe/evaluation/classification/Accuracy.java

```diff
@@ -81,43 +112,79 @@ public String getName() {
 
     @Override
     public final Tuple<List<AggregationBuilder>, List<PipelineAggregationBuilder>> aggs(String actualField, String predictedField) {
-        if (result != null) {
-            return Tuple.tuple(List.of(), List.of());
+        // Store given {@code actualField} for the purpose of generating error message in {@code process}.
+        this.actualField = actualField;
```

Comment:
I don't think we should do this. It is a very unexpected side-effect.

Since there are only two fields `actual` and `predicted` I think it is satisfactory for the error message to simply indicate `Cardinality of actual field is too high`

Reply:
I realize we already do this with result + overall accuracy...I don't particularly like it there either :/

Oh well, I do think it would be better to not store `actualField`.

Reply:
FWIW, the other side-effects of `null` checks. We should at a minimum have a 
```
this.actualField = this.actualField == null ? actualField : this.actualField;
```

Reply:
Some metrics (accuracy, multiclass confusion matrix) need to keep internal non-final state because they require a sequence of aggregations.
So the computation goes like this: aggs(), process(), aggs(), process(), etc. and the intermediate results need to be stored.

I like your suggestion about the "set-once" semantics though. I can even implement it using `SetOnce<>` util class.

Reply:
This is now done.